File: GLibrary-C#/sources/GISocket.cs
using System;
using System.Net;
using System.Net.Sockets;

namespace G
{
    public abstract class GISocket
    {

        public abstract Int32       Send(GString Message);
		public abstract GString     Receive();
        public abstract Socket      GetSocket();

        public static Int32        Send(GISocket Socket, GString ToSend)
        {
	        return (Socket.Send(ToSend));
        }

        public static GString      Receive(GISocket Socket)
        {
	        return (Socket.Receive());
        }
    }
}

File: GLibrary-CPP/sources/GISocket.cpp

#include "GISocket.h"

int			GISocket::Send(GISocket *Socket, const GString &ToSend)
{
	if (Socket)
		return (Socket->Send(ToSend));
	return (0);
}

int			GISocket::Send(GISocket *Socket, void *Data, unsigned int Size)
{
	if (Socket && Size && Data)
		return (Socket->Send(Data, Size));
	return (0);
}

GString		GISocket::Receive(GISocket *Socket)
{
	if (Socket)
		return (Socket->Receive());
	return ("");
}

void		GISocket::Receive(GISocket *Socket, void *Data, unsigned int Size)
{
	if (Socket && Data && Size)
		Socket->Receive(Data, Size);
}

void		GISocket::Close(GISocket *Socket)
{
	if (Socket)
		Socket->Close();
}